libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -429,11 +429,13 @@ - (void)logoutUser:(SFUserAccount *)user
         return;
     }
 
-    if (user.isUserLoggingOut) {
-        [self log:SFLogLevelInfo msg:@"logoutUser: user is already in the process of logout."];
-        return;
+    @synchronized (user) {
```

Comment:
There could be, in the sense that authentication in 5.1 doesn't really know anything about user accounts until credentials have been obtained.  So when you're in the middle of a "new" login, you have no idea what user account will ultimately be impacted by the results.

So I guess you could argue for other guards on the user account.  For instance, if an account is in a "logging out" state, presumably you wouldn't want to switch from there to a "logged in" state, as strange as that scenario may be.